src/StreamJsonRpc/HeaderDelimitedMessageHandler.cs

```diff
@@ -203,50 +206,58 @@ private enum HeaderParseState
         /// <inheritdoc />
         protected override async Task WriteCoreAsync(string content, Encoding contentEncoding, CancellationToken cancellationToken)
         {
-            var sendingBufferStream = new MemoryStream(MaxHeaderElementSize);
+            this.sendingBufferStream.SetLength(0);
```

Comment:
This does seem to release the memory allocated for the earlier buffer according to some online docs. If that's the case and if the reset doesn't honor the initial size that was passed in the constructor earlier, I can see how this can end up causing more memory allocation if it needs to grow its buffer during write operations. Would it be worth just doing a seek to 0 since we already explicitly copy with size.

Maybe the docs are not current though.

Reply:
The docs can be a little confusing. But they don't claim that capacity is ever reduced -- only potentially expanded. In fact in the code I've confirmed that capacity is *not* reduced. So this is an effective way to reuse the existing backing array.